File: examples/museum/oseg_generated/python/BuyMuseumTickets_general_entry.py
from datetime import date, datetime
from pprint import pprint

from openapimuseum_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration()

with ApiClient(configuration) as api_client:
    buy_museum_tickets = models.BuyMuseumTickets(
        ticket_type="general",
        ticket_date=date.fromisoformat("2023-09-07"),
        email="todd@example.com",
        ticket_id=None,
        event_id=None,
    )

    try:
        response = api.TicketsApi(api_client).buy_museum_tickets(
            buy_museum_tickets=buy_museum_tickets,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling Tickets#buy_museum_tickets: %s\n" % e)

File: examples/museum/oseg_generated/ruby/BuyMuseumTicketsGeneralEntry.rb
require "openapimuseum_client"

OpenApiMuseumClient.configure do |config|
end

buy_museum_tickets = OpenApiMuseumClient::BuyMuseumTickets.new
buy_museum_tickets.ticket_type = "general"
buy_museum_tickets.ticket_date = Date.parse("2023-09-07").to_date
buy_museum_tickets.email = "todd@example.com"
buy_museum_tickets.ticket_id = nil
buy_museum_tickets.event_id = nil

begin
    response = OpenApiMuseumClient::TicketsApi.new.buy_museum_tickets(
        buy_museum_tickets,
    )

    p response
rescue OpenApiMuseumClient::ApiError => e
    puts "Exception when calling Tickets#buy_museum_tickets: #{e}"
end
